### DISCUSSION TITLE
Confused about the updated behavior regarding multiple filters for the same field in UI and REST API

### DISCUSSION THREAD

**Discussion by ed-ud** [netbox-community/netbox#12324] [Q&A]
I'm using NetBox v3.4.6.  According to this (from 2019):

https://github.com/netbox-community/netbox/issues/3038

... it sounds the I should be able to query:

/api/ipam/ip-addresses/?tag=foo&tag-bar

... and I should get results for tag=foo OR tag=bar?  However, it seems to behave the old way (tag=foo AND tag=bar).
Same AND behavior happens in the UI at /ipem/ip-addresses/?tag=foo&tag=bar.

When I try the new AND method with /api/ipam/ip-addresses/?tag=foo,bar ... I get:

HTTP 400 Bad Request
Allow: GET, POST, PUT, PATCH, DELETE, HEAD, OPTIONS
Content-Type: application/json
Vary: Accept

{
    "tag": [
        "Select a valid choice. foo,bar is not one of the available choices."
    ]
}

... which is also the old behavior (no support for comma-separated values).  In the UI, it just returns everything as if I didn't specify a filter.

What am I missing?

Thanks,
-- Ed

**Comment by candlerb** [comment]
https://docs.netbox.dev/en/stable/reference/filtering/#filtering-objects

> Generally, passing multiple values for a single parameter will result in a logical OR operation. ... However, a logical AND operation will be used in instances where a field may have multiple values, such as tags. For example, `GET /api/dcim/sites/?tag=foo&tag=bar` will return only sites which have both the "foo" and "bar" tags applied.

That's the current, official documented behaviour.

**Comment by ed-ud** [comment]
So the work done in the case was scrapped?  Unfinished?  It sure seems like the intent of the cases was to make both AND and OR queries possible, even for Tag fields.  Someone even noted in the case that the documentation wasn't indicating the enhanced behavior.

**Reply by candlerb** [reply]
It is how it is. See #11703 - it appears to be limited to what Django can do.
